cranelift/isle/isle/src/parser.rs

```diff
@@ -338,17 +346,12 @@ impl<'a> Parser<'a> {
 
     fn parse_extern(&mut self) -> Result<Extern> {
         let pos = self.pos();
-        if self.is_sym_str("constructor") {
-            self.symbol()?;
-
+        if self.eat_sym_str("constructor")? {
             let term = self.parse_ident()?;
             let func = self.parse_ident()?;
             Ok(Extern::Constructor { term, func, pos })
-        } else if self.is_sym_str("extractor") {
-            self.symbol()?;
-
-            let infallible = if self.is_sym_str("infallible") {
-                self.symbol()?;
+        } else if self.eat_sym_str("extractor")? {
+            let infallible = if self.eat_sym_str("infallible")? {
```

Comment:
This can just be `let infallible = self.eat_sym_str("infallible")?;` now, in yet another demonstration of why your suggested change is a good idea.

Reply:
Indeed. Fixed a couple of other cases like this too.